components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/WorkflowsApiServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+import org.wso2.carbon.apimgt.core.util.APIMgtConstants;
+import org.wso2.carbon.apimgt.core.workflow.WorkflowExecutorFactory;
+import org.wso2.carbon.apimgt.rest.api.common.dto.ErrorDTO;
+import org.wso2.carbon.apimgt.rest.api.common.util.RestApiUtil;
+import org.wso2.carbon.apimgt.rest.api.store.NotFoundException;
+import org.wso2.carbon.apimgt.rest.api.store.WorkflowsApiService;
+import org.wso2.carbon.apimgt.rest.api.store.dto.WorkflowDTO;
+
+import java.util.HashMap;
+
+import javax.ws.rs.core.Response;
+
+public class WorkflowsApiServiceImpl extends WorkflowsApiService {
+
+    private static final Logger log = LoggerFactory.getLogger(WorkflowsApiServiceImpl.class);
+
+    @Override
+    public Response workflowsUpdateWorkflowStatusPost(String workflowReferenceId, WorkflowDTO body)
+            throws NotFoundException {
+        String username = RestApiUtil.getLoggedInUsername();
+        try {
+
+            APIStore apiStore = RestApiUtil.getConsumer(username);
+            Workflow workflow = apiStore.retrieveWorkflow(workflowReferenceId);
+            if (workflow == null) {
+                String errorMessage = "Workflow entry not found for: " + workflowReferenceId;
+                APIMgtResourceNotFoundException e = new APIMgtResourceNotFoundException(errorMessage,
+                        ExceptionCodes.WORKFLOW_NOT_FOUND);
+                HashMap<String, String> paramList = new HashMap<String, String>();
+                paramList.put(APIMgtConstants.ExceptionsConstants.WORKFLOW_REF_ID, workflowReferenceId);
+                ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getErrorHandler(), paramList);
+                log.error(errorMessage, e);
+                return Response.status(e.getErrorHandler().getHttpStatusCode()).entity(errorDTO).build();
+            } else if (WorkflowStatus.APPROVED == workflow.getStatus()) {
+                String errorMessage = "Workflow is already in complete state";
+                APIMgtResourceNotFoundException e = new APIMgtResourceNotFoundException(errorMessage,
+                        ExceptionCodes.WORKFLOW_COMPLETED);
+                HashMap<String, String> paramList = new HashMap<String, String>();
+                paramList.put(APIMgtConstants.ExceptionsConstants.WORKFLOW_REF_ID, workflowReferenceId);
+                ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getErrorHandler(), paramList);
+                log.error(errorMessage, e);
+                return Response.status(e.getErrorHandler().getHttpStatusCode()).entity(errorDTO).build();
+
+            } else {
+                WorkflowExecutor workflowExecutor = WorkflowExecutorFactory.getInstance()
+                        .getWorkflowExecutor(workflow.getWorkflowType());
+
+                if (body == null) {
+                    RestApiUtil.handleBadRequest("Request payload is missing", log);
+                }
+
+                if (body.getDescription() != null) {
```

Comment:
Here checking whether getDescription() method exists to prevent NPE when calling the setter (body object is generated by incoming json payload). wouldn't be a problem for an empty string